### PR TITLE
Implement proof of concept of pip package bulding & repo versioning

### PR DESCRIPTION
**Pull request by nekoffski**

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+[build-system]
+requires = ["hatchling"]
+build-backend = "hatchling.build"
+
+[project]
+name = "foxlator-lib"
+version = "0.0.2"
+authors = [
+  { name="Witold Haza", email="witekha3@gmail.com" },
```
Comment (witekha3) [inline pyproject.toml]: witekhaface@gmail.com

File: bin/install-deps.sh [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+echo "$ - installing python interpreter and necessary modules"
+sudo add-apt-repository ppa:deadsnakes/ppa -y
+sudo apt-get update
+sudo apt-get install -y python3.10 python3.10-distutils python3.10-venv
+
+echo "$ - installing pip"
+curl -sS https://bootstrap.pypa.io/get-pip.py | python3.10
+
+python3.10 -m pip install --upgrade build
+python3.10 -m pip install --upgrade twine
```
Comment (witekha3) [inline bin/install-deps.sh]: We can create a requirements.txt file and force the installation of libs here

**Review by witekha3 — CHANGES_REQUESTED** [review]
I know we've discussed that we can use Python 3.10, but I'm wondering if we should make it more "Python version independent" (with limitations, of course). Maybe we should retrieve the Python version from env and use 3.10 as the default if we don't have it? This should make things easier in case we ever need to change version or test a lib on another version

**Comment by nekoffski** [comment]
> I know we've discussed that we can use Python 3.10, but I'm wondering if we should make it more "Python version independent" (with limitations, of course). Maybe we should retrieve the Python version from env and use 3.10 as the default if we don't have it? This should make things easier in case we ever need to change version or test a lib on another version

File: pyproject.toml (hunk 1 above)
Comment (nekoffski) [reply pyproject.toml]: Updated.

**Comment by nekoffski** [comment]
> I know we've discussed that we can use Python 3.10, but I'm wondering if we should make it more "Python version independent" (with limitations, of course). Maybe we should retrieve the Python version from env and use 3.10 as the default if we don't have it? This should make things easier in case we ever need to change version or test a lib on another version

Personally, I wouldn't say I like this idea. If we want to support some limitations then we need to stick to the lowest possible version of Python that we want to support, and so we could just specify that version in the config file. That doesn't make sense to me.

File: bin/install-deps.sh (hunk 2 above)
Comment (nekoffski) [reply bin/install-deps.sh]: Make sense.

**Review by witekha3 — APPROVED** [review]
LGTM

**Pull request merged**